full-service/src/service/sync.rs

```diff
@@ -122,8 +122,9 @@ pub fn sync_all_accounts(
         { Account::list_all(conn, None, None).expect("Failed getting accounts from database") };
 
     for account in accounts {
-        // If there are no new blocks for this account, don't do anything.
-        if account.next_block_index as u64 > num_blocks - 1 {
+        // If there are no new blocks for this account, or we are in offline mode (so
```

Comment:
Being in offline mode doesn't mean there are no blocks, as a user can manually load a recent ledger_db from an online machine, which would trigger this. This shouldn't affect the logic here, but I feel justifies changing the comment.

Reply:
Also, this check for if num_blocks == 0 might be better before this loop through the accounts, since it would apply to the whole wallet (basically this is a redundant check)